modules/node_groups/templates/userdata.sh.tpl

```diff
@@ -1,9 +1,20 @@
 #!/bin/bash -e
+%{ if length(ami_id) == 0 || ami_is_eks_optimized ~}
 
-# Allow user supplied pre userdata code
+sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
+%{endif ~}
+%{if length(pre_userdata) > 0 ~}
+
+# User supplied pre userdata code
 ${pre_userdata}
+%{endif ~}
+%{ if length(ami_id) > 0 && ami_is_eks_optimized ~}
 
-sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
-%{ if run_bootstrap_script }
-    /etc/eks/bootstrap.sh ${cluster_name}
-%{ endif }
+# Set variables
+API_SERVER_URL=${cluster_endpoint}
+B64_CLUSTER_CA=${cluster_ca}
+K8S_CLUSTER_DNS_IP=172.20.0.10
```

Comment:
I found one issue as this should be variable https://github.com/terraform-aws-modules/terraform-aws-eks/blob/253f927465ccac36bcadefcfb1fd0e39b6244d73/main.tf#L28

Reply:
@daroga0002 I think you're right. I'm currently on vacation, so can't make this change until I'm back next week. If you can modify my PR that's fine. Otherwise setting the optimised variable to false would let this to be set by providing the whole user data, I could then update the other PR to make this customisable as it covers this type of behaviour?

Reply:
np, we can wait, have a great vacations 😄

Reply:
@daroga0002 this should be ready to merge.

Reply:
@stevehipwell are you sure you added change as I still see 
![image](https://user-images.githubusercontent.com/20227477/135910579-ca777295-51a6-4661-81d7-76ed43bfa2e5.png)

Reply:
@daroga002 sorry, manic day in the office. I only rebased, I'll add the changes first thing in the morning.

Reply:
@daroga0002 I've updated the userdata template to be simpler and so that all custom AMIs use the same initial logic for setting the variables (this is designed ready to support #1577). After checking the _bootstrap.sh_ code I've removed the manual setting of `K8S_CLUSTER_DNS_IP` as _bootstrap.sh_ will handle this correctly; #1577 will enable this to be fully customisable.